extension/chrome/settings/setup/setup-render.ts

```diff
@@ -46,7 +48,17 @@ export class SetupRenderModule {
       await this.view.submitPublicKeysAndFinalizeSetup({ submit_all: tmp_submit_all, submit_main: tmp_submit_main });
       await this.renderSetupDone();
     } else if (this.view.orgRules.mustAutoImportOrAutogenPrvWithKeyManager()) {
-      await this.view.setupWithEmailKeyManager.setupWithEkmThenRenderSetupDone();
+      if (this.view.orgRules.mustAutogenPassPhraseQuietly() && this.view.orgRules.forbidStoringPassPhrase()) {
+        const notSupportedErr = 'Combination of org rules not valid: PASS_PHRASE_QUIET_AUTOGEN cannot be used together with FORBID_STORING_PASS_PHRASE.';
+        await Ui.modal.error(notSupportedErr);
+        window.location.href = Url.create('index.htm', { acctEmail: this.view.acctEmail });
+        return;
+      }
+      if (this.view.orgRules.userMustChoosePassPhraseDuringPrvAutoimport()) {
+        this.displayBlock('step_2_ekm_choose_pass_phrase');
```

Comment:
There is small possibility that subsequent key retrieval will fail with a network error exception 
`FlowCrypt Email Key Manager at ... is down, please inform your network admin.` AFTER the passphrase is enter. 
I personally hate sites that make user enter some input too early so that the input will get lost under some circumstances, like intermittent network failures. However, moving passphrase entering to a later stage will considerably increase the code complexity. So are we going to go this way to improve UX, @tomholub ?

Reply:
I have a method that offers the user to retry the flaky step until the step succeeds, without losing variables like pass phrase.

You can see it in action here:

```ts
    const storePrvOnKm = async () => this.view.keyManager!.storePrivateKey(this.view.idToken!, KeyUtil.armor(decryptablePrv), pubArmor);
    await Settings.retryUntilSuccessful(storePrvOnKm, 'Failed to store newly generated key on FlowCrypt Email Key Manager');
```

This could allow you to ask for pass phrase early but still have good UX when things go wrong.

Reply:
This is already in use in `autoGenerateKeyAndStoreBothLocallyAndToEkm()` stage.
The current implementation for the "get" stage failure `const { privateKeys } = await this.view.keyManager!.getPrivateKeys(this.view.idToken!);`
is to tell the user to "please inform your network admin." I see no point in retries here.
We'd better move the passphrase input screen after the successful completion of "get the key from ekm" stage.

Reply:
There typically is a point in retrying, but it depends on the error. Network issues, server down and 5xx codes are retriable - and I think these would be the most common.